packages/main/src/plugin/telemetry/telemetry-api.ts

```diff
@@ -0,0 +1,28 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+export interface TelemetryRule {
+  // start of telemetry event id
+  event: string;
+  // ratio to send events, e.g. 0.5
+  ratio?: number;
+  // disable this event entirely
+  disabled?: boolean;
+  // limit sending to 'dailyPerInstance'
+  frequency?: string;
```

Comment:
can't we have an enum here for better readability

Reply:
Fixed in new commit.